var/spack/repos/builtin/packages/tfel/package.py

```diff
@@ -155,12 +155,22 @@ class Tfel(CMakePackage):
     depends_on("python", when="+python_bindings", type=("build", "link", "run"))
     depends_on("py-numpy", when="+python_bindings", type=("build", "link", "run"))
 
-    # As boost+py has py runtime dependency, boost+py needs types link and run as well:
-    depends_on(
-        "boost+python+numpy+exception+container",
-        when="+python_bindings",
-        type=("build", "link", "run"),
-    )
+    with when("@5.1:"):
+        depends_on("py-pybind11", when="+python_bindings", type=("build", "link", "run"))
+
+    with when("@2.0.4:5.0.99"):
+        depends_on(
+            "boost+python+numpy+exception+container",
+            when="+python_bindings",
+            type=("build", "link", "run"),
+        )
+
+    with when("@rliv1.2:rliv5.0"):
+        depends_on(
+            "boost+python+numpy+exception+container",
+            when="+python_bindings",
+            type=("build", "link", "run"),
+        )
```

Comment:
(Optional comment on style) This is probably better factored by grouping on:
```python
with when("+python_bindings"):
```
since, in that case, you'll see under the same block the different dependencies needed to activate the same functionality.

Reply:
Yes, I see. I need to make a minor release of `mgis` soon, I'll do that when I'll update the packages, if you don't mind.